src/modules/ctc/src/ctc.cc

```diff
@@ -64,6 +64,53 @@ types::Error Ctc::UpdateSuggestedSpeedAndAuthority(const types::TrainId train_id
     return error;
 }
 
+types::Error Ctc::SetBlockStates(const types::TrackId track, const std::vector<types::BlockState> &block_states)
+{
+    types::Error   error     = types::Error::ERROR_INVALID_FORMAT;
+    types::TrackId ctc_track = GetTrack();
+    if (track == ctc_track)
+    {
+        for (const types::BlockState &block_state : block_states)
+        {
+            // Update block states in private data memer blocks_ which stores all blocks information
+            std::shared_ptr<types::Block> block_ptr = std::make_shared<types::Block>();
+            if (GetBlockPoitnerById(block_state.block, block_ptr) == types::Error::ERROR_NONE)
+            {
+                block_ptr->occupied = block_state.occupied;
+                if (block_state.track_failure == true)
+                {
+                    failure_blocks_.push_back(block_state.block);
+                }
+                error = types::Error::ERROR_NONE;
```

Comment:
Not sure the logic here makes sense.  If the intent is to return an error if one or more of the `block_states` does not have a matching block in the CTC's list of blocks, `ERROR_NONE` will still be returned as long as at least a single block matches even if several others are not found.

Reply:
Fixed by adding else case iterator can't find block_id then set error to invalid block id